src/common.jl

```diff
@@ -23,7 +23,7 @@ decentralize(x::AbstractMatrix, m::AbstractVector) = (isempty(m) ? x : x .+ m)
 fullmean(d::Int, mv::Vector{T}) where T = (isempty(mv) ? zeros(T, d) : mv)::Vector{T}
 
 preprocess_mean(X::AbstractMatrix{T}, m) where T<:Real =
-    (m === nothing ? vec(mean(X, dims=2)) : m == 0 ? T[] :  m)
+    (m == nothing ? vec(mean(X, dims=2)) : m == 0 ? T[] :  m)::Vector{T}
```

Comment:
I removed it this constraint before in #117. Why do you need it?

Reply:
Sorry about that. Yeah we don't need it. I'll fix it

Reply:
@wildart Thanks for taking the time to review my PR